### PR TITLE
introduce next generation test suites

### PR DESCRIPTION
**Pull request by xueqzhan**
## Summary by CodeRabbit

* **Documentation**
  * Defined metadata-based suite classification using lifecycle, criticality, execution mode, and shard tags.
  * Documented lifecycle stages from Draft through Stable, including promotion, graduation, and longevity requirements.
  * Clarified suite membership rules, metadata validation, shard registration, and automated shard assignment.
  * Documented tag-promotion conflict handling and consistent lifecycle reporting.
  * Clarified conformance shard naming, approvals, infrastructure requirements, and suite-name uniqueness.
  * Recorded downstream minimal-suite membership criteria as unresolved.

### CODE REVIEW

File: enhancements/testing/test-suites.md [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+  I can stabilize new tests without risking release signal.
+* As a CI infrastructure engineer, I want explicitly sharded suites
+  so that I can schedule jobs predictably in resource-constrained
+  environments.
+* As a component team lead, I want spot-check suites for features
+  requiring uncommon cluster configurations so that specialized
+  tests do not pollute the main conformance signal.
+* As a member of the quality staff engineer, I want shard runtimes
+  balanced within 10% of mean so that CI pipelines complete in
+  predictable and roughly equal time windows.
```
Comment (miyadav) [inline enhancements/testing/test-suites.md:77]: What values will take as starting point ? how we will get the statistics , so those adjustments will be ongoing ?

File: enhancements/testing/test-suites.md [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+#### Graduating a Test to Stable
+
+1. After GA + 1 release, the QSE engineer changes the feature suite's
+   `Parents` from `openshift/active-*` to `openshift/stable-NN`.
+2. The QSE engineer selects the appropriate shard based on current
+   shard runtimes.
+3. The test now runs permanently in the `stable` suite with a >= 99.5%
+   pass-rate expectation.
+
+#### Setting Up a Spot-Check Suite
```
Comment (miyadav) [inline enhancements/testing/test-suites.md:199]: Shall we define life cycle for spot-check ,suites as well , I mean in terms of removing them after a while  , if there is a surge of such jobs and we want to keep a check on total number of such configurations at a time ?

File: enhancements/testing/test-suites.md [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+   (serial).
+3. The test author marks the test as `Informing()` so failures are
+   non-blocking during stabilization.
+4. The test is picked up in CI through the OTE extension binary
+   discovery mechanism and runs in the `active` suite for 2–3
+   sprints.
+
+#### Promoting a Test to Blocking
+
+1. After 2–3 sprints, the QSE engineer reviews the test's pass rate.
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md]: This really should be an agent, there's nothing here that can't be automated easily. We could do a prow job on a weekly or sprintly schedule, scan the informing tests, check the pass rate, submit the PR to remove it. This would keep QSE's out of the path from the get-go.

File: enhancements/testing/test-suites.md [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+2. If the pass rate is >= 99%, the QSE engineer removes the
+   `Informing()` annotation, making the test blocking within the
+   `active` suite.
+3. The test remains blocking in `active` until GA + 1 release.
+
+#### Graduating a Test to Stable
+
+1. After GA + 1 release, the QSE engineer changes the feature suite's
+   `Parents` from `openshift/active-*` to `openshift/stable-NN`.
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md]: Agent again, could be part of the same one described above.

File: enhancements/testing/test-suites.md [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+2. The QSE engineer selects the appropriate shard based on current
+   shard runtimes.
+3. The test now runs permanently in the `stable` suite with a >= 99.5%
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md]: 99.5% seems off here, it may just be component readiness at that point, and it would be holding a roughly 95% pass rate requirement. (assuming it was around 100 when it promoted, and our comp readiness parameters allow -5%)

File: enhancements/testing/test-suites.md [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+
+#### Setting Up a Spot-Check Suite
+
+1. The test author identifies tests requiring uncommon cluster
+   configurations (e.g., etcd scaling, realtime nodes, external OIDC).
+2. The test author creates a suite under `openshift/spot-check/<feature>`
+   with dedicated CI jobs using specialized cluster configs.
+3. During development, the spot-check job runs approximately 2x daily
+   (~14+ runs/week) for pass-rate analysis.
+4. Post-GA + 1 release, the frequency drops to approximately 1x/month.
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:208]: etcd-scaling preferred to keep it once a week, and there will hopefully one day be an automatic retries mechanism if it fails. This is ok as written though, the schedule would likely be 1x a month still.

File: enhancements/testing/test-suites.md [hunk 7]
```diff
@@ -0,0 +1,379 @@
...
+4. The test is picked up in CI through the OTE extension binary
+   discovery mechanism and runs in the `active` suite for 2–3
+   sprints.
+
+#### Promoting a Test to Blocking
+
+1. After 2–3 sprints, the QSE engineer reviews the test's pass rate.
+2. If the pass rate is >= 99%, the QSE engineer removes the
+   `Informing()` annotation, making the test blocking within the
+   `active` suite.
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md]: Could the enhancement call out how we can automate locating tests? I think there's a piece of work here hopefully TRT could take which is a more robust test database, which knows where the source code is. This would give us a central index to scan for tests that are informing or due for promotion, which we have already, but we don't necessarily know where they live.

File: enhancements/testing/test-suites.md [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+#### Single-node Deployments or MicroShift
+
+No additional resource consumption is introduced by this proposal.
+
+#### OpenShift Kubernetes Engine
+
+No dependency on features excluded from OKE. The suite hierarchy
+applies to OKE testing in the same way as OCP.
+
+### Implementation Details/Notes/Constraints
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:240]: Can you add some sections here for some complicated scenarios. 

What happens with tests coming in from an upstream repo and landing in our core suites, like kube does. I see about 340 tests from Suite:k8s, but they all seem to be marked conformance/minimal. Are there other upstream tests we take or just these minimal ones? 

Are there other repos with tests coming via upstream we can't just control and move around? I guess by using OTE, likely we have the ability to categorize as we want.

File: enhancements/testing/test-suites.md [hunk 9]
```diff
@@ -0,0 +1,379 @@
...
+
+#### Suite Hierarchy
+
+The new hierarchy is built on top of the OTE APIs:
+
+| Suite | Description |
+|-------|-------------|
+| `openshift/conformance/parallel/minimal` | Core smoke tests only, parallel execution |
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:250]: Is openshift/conformance/parallel a thing (without minimal?) Where do those tests land for a suite?

File: enhancements/testing/test-suites.md [hunk 10]
```diff
@@ -0,0 +1,379 @@
...
+| `openshift/conformance/serial/minimal` | Core smoke tests only, serial execution |
+| `openshift/active-01` | Recent features, parallel, shard 1 |
+| `openshift/active-serial-01` | Recent features, serial, shard 1 |
+| `openshift/stable-01` | Graduated tests, parallel, shard 1 |
+| `openshift/stable-02` | Graduated tests, parallel, shard 2 |
+| `openshift/stable-serial-01` | Graduated tests, serial, shard 1 |
+| `openshift/spot-check/<feature>` | Uncommon cluster configs |
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:257]: Is spot-check important for the suite name, is it purely informational to a human reader or does it have any functionality? 

Maybe someday automatic management of the prow jobs for them... Hmm..

File: enhancements/testing/test-suites.md [hunk 11]
```diff
@@ -0,0 +1,379 @@
...
+| Phase | Suite | Pass Rate Threshold | Duration |
+|-------|-------|---------------------|----------|
+| New / Stabilizing | `active` with `Informing()` | Failures non-blocking | 2–3 sprints |
+| Blocking | `active` (remove `Informing()`) | >= 99% | Until GA + 1 release |
+| Graduated | `stable-NN` | >= 99.5% | Permanent |
+| Specialized | `spot-check/<feature>` | N/A (own job) | Own cadence |
+
+#### Suite Composition with OTE APIs
+
+Feature suites compose into parent suites via the `Parents` field:
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:313]: This could be used by ai in the future to target jobs to feature suites for quicker presubmit testing. I like it.

File: enhancements/testing/test-suites.md [hunk 12]
```diff
@@ -0,0 +1,379 @@
...
+  2. Move tests between shards by adjusting `SHARD-NN` labels.
+  3. Create new shards when existing ones cannot be rebalanced below
+     runtime thresholds.
+
+#### Test Requirements Checklist
+
+Before assigning any test to a suite, ensure the following:
+
+- Has `[Jira:Component]` tag or `ci-test-mapping` entry.
+- Has `[sig-XYZ]` tag.
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md]: I think we just want the Jira component now, this method is deprecated.

File: enhancements/testing/test-suites.md (hunk 8 above)
Comment (dgoodwin) [reply enhancements/testing/test-suites.md]: And another (sorry), could you spell out the plan for jobs, what happens to the jobs that exist today, such as aws-ovn, aws-ovn-upgrade, aws-ovn-serial. What suite will these be running? Will jobs need to be mass renamed?

File: enhancements/testing/test-suites.md [hunk 13]
```diff
@@ -0,0 +1,379 @@
...
+
+### Risks and Mitigations
+
+**Risk**: Component teams may resist shrinking conformance and
+relocating tests to active/stable suites.
+**Mitigation**: The migration can be phased, starting with new tests
+using the new hierarchy while existing conformance tests are migrated
+gradually. Clear documentation and tooling will ease the transition.
+
+**Risk**: Explicit sharding may become unbalanced over time as tests
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:670]: I'd expand on this risk to include "and require manual maintenance", with the mitigation including an agent that handles all the PRs for rebalancing and promotion.

File: enhancements/testing/test-suites.md [hunk 14]
```diff
@@ -0,0 +1,379 @@
...
+with periodic reviews by QSE engineers.
+
+**Risk**: Spot-check suites with low run frequency may go stale
+without detection.
+**Mitigation**: Component Readiness flags trigger if no pass occurs
+in 30 days, ensuring stale suites are surfaced automatically.
+
+### Drawbacks
+
+- Explicit sharding requires manual maintenance compared to
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md]: Can go away, though PRs will have to be approved. 

We have a developing need for PRs where teams have X days to respond, otherwise an architect or staff eng may merge it. This could be considered here, a PR to rebalance shards or promote tests does not really need team involvement.

File: enhancements/testing/test-suites.md [hunk 15]
```diff
@@ -0,0 +1,379 @@
...
+- The migration from the flat suite model to the hierarchical model
+  will require coordination across many component teams.
+- Additional suite definitions increase the complexity of CI job
+  configuration in `openshift/release`.
+
+## Open Questions [optional]
+
+1. What is the exact criteria for a test to remain in the minimal
+   conformance suites versus being moved to `stable`?
+2. Should shard-balancing be automated in the future, and if so, what
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md]: I think this would be so time consuming to do today manually it would never happen, we should just skip that phase and go straight to the agent from the get-go. I really think it's doable with precedent from stbenjam and co. It also closes the gap we have for informing tests getting left there forever, in both good and bad states.

File: enhancements/testing/test-suites.md [hunk 16]
```diff
@@ -0,0 +1,379 @@
...
+
+No additional resource consumption is introduced by this proposal.
+
+#### OpenShift Kubernetes Engine
+
+No dependency on features excluded from OKE. The suite hierarchy
+applies to OKE testing in the same way as OCP.
+
+### Implementation Details/Notes/Constraints
+
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:241]: What's the plan for suites in test names like we do in origin.

File: enhancements/testing/test-suites.md [hunk 17]
```diff
@@ -0,0 +1,379 @@
...
+#### Suite Hierarchy
+
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:243]: And yet another, could we discuss how and where the jobs will be used, and potential cost concerns or improvements. I was envisioning that minimal is used on prs and payloads. Active and stable regressions are caught in component readiness. (I'm a little more flexible on active, maybe that goes to payloads as well) 

I don't see any path through this where cost is lowered, but I'm not sure. The most commonly run jobs will be minimal, they'll be shorter and faster. Active will run less often, and stable even less than active. The 40 min overhead of getting a cluster though may negate the savings, but I'm really not sure.

File: enhancements/testing/test-suites.md [hunk 18]
```diff
@@ -0,0 +1,379 @@
...
+No additional resource consumption is introduced by this proposal.
+
+#### OpenShift Kubernetes Engine
+
+No dependency on features excluded from OKE. The suite hierarchy
+applies to OKE testing in the same way as OCP.
+
+### Implementation Details/Notes/Constraints
+
+#### Suite Hierarchy
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:242]: And another (sorry), could you spell out the plan for jobs, what happens to the jobs that exist today, such as aws-ovn, aws-ovn-upgrade, aws-ovn-serial. What suite will these be running? Will jobs need to be mass renamed?

**Review by dgoodwin — COMMENTED** [review]
Fantastic job on the enhancement, really lays out the plan simply. Key additions I'd like to see are an agent to handle the maintenance of tests and suites, and a number of edge cases and the plan for jobs.

File: enhancements/testing/test-suites.md (hunk 2 above)
Comment (miyadav) [reply enhancements/testing/test-suites.md:199]: Also , do we want any approval process for this or keep it unchecked ? So as to give predictable environment for such tests.?

File: enhancements/testing/test-suites.md [hunk 19]
```diff
@@ -0,0 +1,838 @@
...
+- **Upstream Kubernetes tests**: minimal membership is selected
+  mechanically by the upstream `[Conformance]` label. Tests carrying
+  that label are placed in `conformance/*/minimal`; the mapping is
+  applied centrally rather than judged test-by-test (see
+  [Upstream and Externally-Sourced Tests](#upstream-and-externally-sourced-tests)).
+- **Downstream (OpenShift-authored) tests**: there is no equivalent
+  established signal. Deciding which downstream tests deserve to be
+  labelled/treated as conformance-minimal still requires evaluation,
+  likely a manual review per component to begin with. Defining an
+  objective, repeatable rule for this is tracked as an open question.
```
Comment (miyadav) [inline enhancements/testing/test-suites.md:288]: May be below can be thought of (just  my thoughts , needs broader audience to suggest) - 
 
  - Test duration < 2 minutes
  - Covers a GA feature that would break 80%+ of clusters if removed
  - No external dependencies beyond core k8s API 
  - Has been stable (>99%) for 2+ releases
  - Reviewed by architect

File: enhancements/testing/test-suites.md [hunk 20]
```diff
@@ -0,0 +1,838 @@
...
+  paired with an automatic retry-on-failure mechanism once available).
+- **Maintenance Phase**: Post-GA + 1 release, frequency drops to
+  ~1x/month.
+- **Health monitoring**: Spot-check suite health is monitored by
+  Component Readiness. The lifecycle agent does not manage spot-check
+  suites or jobs.
+- **Monitoring**: Monitortests collect data but avoid generating
+  failure JUnits to prevent alerts on known specialized configs.
+
+#### Shard Balancing
```
Comment (miyadav) [inline enhancements/testing/test-suites.md:426]: Nit , do we want to define the max shard as well , to control unexpected situation triggering creation of too many shards per suite ?

File: enhancements/testing/test-suites.md [hunk 21]
```diff
@@ -80,15 +96,20 @@ and balance shards.
    uncommon cluster configurations.
 6. Establish a shard-balancing workflow using Sippy data to keep all
    shards within 10% of mean runtime.
+7. Provide an automation agent (a scheduled prow job) that drives
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:99]: @stbenjam @enxebre in the context of an agent monitoring test lifecycle, moving informing tests to blocking when ready, promoting tests to stable suites after a few releases, rebalancing shards, how would you advice we implement it? At the time we were thinking ai-helpers skills and a prow job, is this approach being superseded by chai scheduled tasks? Are ai-helpers recommended or just pure chai skills/plugins/etc.

File: enhancements/testing/test-suites.md [hunk 22]
```diff
@@ -111,37 +139,75 @@ for suite management and test graduation.
...
+   engineer) approves the pull request.
+4. The test remains blocking in `active` until GA + 1 release.
+
+Promotion is one-way. Once a test is made blocking, it is never
+demoted back to `Informing()` and it is never moved to a lower-signal
+suite. If its pass rate later degrades, that is treated as a
+**regression to be fixed**, not as grounds for quarantine or downgrade.
+The agent surfaces the degradation (and the owning component is expected
+to fix it) rather than relaxing the test's lifecycle state. See
+[Pass-Rate Calculation](#pass-rate-calculation).
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:168]: fyi @ironcladlou just regarding your inquiry earlier this week.

File: enhancements/testing/test-suites.md (hunk 19 above)
Comment (dgoodwin) [reply enhancements/testing/test-suites.md]: This is a good starting point.

File: enhancements/testing/test-suites.md [hunk 23]
```diff
@@ -188,74 +256,256 @@ The new hierarchy is built on top of the OTE APIs:
...
 
 ```go
-// Define a feature suite that composes into active
-ext.AddSuite(e.Suite{
-    Name:    "mycomponent/feature-x",
-    Parents: []string{"openshift/active-01"},
-})
+// Label tests for suite membership. Qualifiers use the documented
+// OTE CEL contract: test.labels.has("...") / test.name.contains("...").
+specs.Select(et.NameContains("[sig-foo] my test")).AddLabel("MY-FEATURE")
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:318]: Is the actual assignment for a test not defined with the test? If I have a test defined one place, and then what would surely be a very large set of lines like the above (thousands) defined in another place, assigning tests based on pattern matching, wouldn't it become very difficult to actually see what suite/job a test is in at any given point? I'd expected that tests would all have their current status defined directly with the test so it was easily visible. 

Or would all tests for a feature be promoted together?

File: enhancements/testing/test-suites.md [hunk 24]
```diff
@@ -188,74 +256,256 @@ The new hierarchy is built on top of the OTE APIs:
...
+// OTE merges Parents additively across extension binaries, so leaving
+// both advertisements in place would make the test a member of both
+// suites. A parallel test maps to an explicit stable-NN shard; a
+// serial test maps to an explicit stable-serial-NN shard.
 ext.AddSuite(e.Suite{
     Name:       "mycomponent/feature-x",
-    Qualifiers: []string{`labels.exists(l, l=="MY-FEATURE")`},
-    Parents:    []string{"openshift/active-01"},
+    Qualifiers: []string{`test.labels.has("MY-FEATURE")`},
+    Parents:    []string{"openshift/stable-01"}, // was openshift/active-01
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:336]: This makes me think suites are promoted, not tests themselves? So a test would be promoted only if ALL tests in it's feature suite are 99% plus and have been there for so many releases GA'd? What if someone was adding a new test, to a feature suite that has been promoted?

File: enhancements/testing/test-suites.md [hunk 25]
```diff
@@ -188,74 +256,256 @@ The new hierarchy is built on top of the OTE APIs:
...
+Reliable automation requires knowing which tests are `Informing()`,
+which are eligible for promotion, and — importantly — where their source
+lives so a PR can be opened against the right repository. Today the CI
+signal knows a test's name and state but not necessarily its source
+location. This enhancement depends on (and TRT is expected to provide) a
+more robust test database that maps each test to its owning repository
+and source, giving the agent a central index to scan. Until that index
+exists, the agent can operate on the subset of tests whose source
+location is already resolvable via `ci-test-mapping` / the OTE
+`source` component.
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:476]: cc @neisw @smg247 regarding this expectation for TRT. I don't think I've filed jiras for this yet.

File: enhancements/testing/test-suites.md [hunk 26]
```diff
@@ -265,6 +515,150 @@ Before assigning any test to a suite, ensure the following:
...
+- **Labelling is done centrally, not inline.** Because we do not want to
+  edit vendored upstream test source, the shard/minimal labels are
+  assigned by a dedicated function that walks the imported specs and
+  applies labels from a rules table (see the example in
+  [Mandatory Labelling](#mandatory-labelling)). This mirrors how
+  openshift/kubernetes already classifies these tests today — on `master`
+  via the `k8s-tests-ext` OTE extension binary's centralized label and
+  qualifier tables, and on release branches via the legacy
+  `annotate.go`/`rules.go` mechanism.
+- **We retain the ability to categorize.** Because everything flows
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:565]: love this.

File: enhancements/testing/test-suites.md [hunk 27]
```diff
@@ -173,21 +200,21 @@ Graduation is also performed by the lifecycle agent.
...
+   always chosen.
+2. The agent opens a single pull request that flips the test's
+   `LifeCycle` tag from `Blocking` to `Stable` (moving it from the
+   `active` selection to the `stable` selection) and assigns its shard
+   tag, so metadata and shard membership stay consistent. Because the
+   `active`/`stable` suites are selected by the single-valued `LifeCycle`
+   tag, changing that one value atomically moves the test; there is no
+   window in which it belongs to both. This consistency requirement is
+   why every test must carry the required metadata tags (enforced in CI;
+   see [Mandatory Metadata](#mandatory-metadata)) — an untagged test
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:217]: I'm still unsure if we want Stable to be reflected in the lifecycle. To me they're separate concepts, lifecycle is how the test behaves within a job. The other is suite selection. It's confusing as stable very much feels like lifecycle, and related to blocking, but I'm just not sure it's right to conflate them. Stable then becomes duplicated between the lifecycle and the suite name.

File: enhancements/testing/test-suites.md [hunk 28]
```diff
@@ -239,22 +267,86 @@ applies to OKE testing in the same way as OCP.
...
+Suite membership is driven by structured key/value **metadata tags**
+rather than opaque string labels. Tags are a first-class OTE feature:
+they are set with `AddTag({"key":"value"})` and queried in CEL suite
+qualifiers as `test.tags.<key>=="<value>"` (see
+[openshift-tests-extension](openshift-tests-extension.md)). This
+enhancement defines a single ordered lifecycle axis plus one overlay:
+
+| Key | Valid values | Meaning |
+|-----|--------------|---------|
+| `LifeCycle` | `Draft`, `Informing`, `Blocking`, `Stable` | The one ordered axis describing where a test sits from creation to permanent graduation. `Draft` → in no suite; `Informing` → runs non-blocking in `active`; `Blocking` → runs and gates in `active`; `Stable` → graduated, gates permanently in `stable`. |
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:281]: To my above point, this opens the door to confusing things, can there be an informing test in stable, can there be a stable test in the active suite.

File: enhancements/testing/test-suites.md [hunk 29]
```diff
@@ -239,22 +267,86 @@ applies to OKE testing in the same way as OCP.
...
+  - `Draft` — the test is in **no** suite (not `active`, `stable`, or
+    `minimal`) and does not run. It is a work-in-progress the owner has
+    not yet released into CI.
+  - `Informing` — runs in the `active` suites, failures non-blocking.
+  - `Blocking` — runs in the `active` suites and gates.
+  - `Stable` — graduated (after GA + 1); runs in the `stable` suites and
+    gates permanently.
+- **Suite membership derives from `LifeCycle`:** `active` selects
+  `Informing` or `Blocking`; `stable` selects `Stable`. Suite qualifiers
+  exclude `Draft` implicitly (it matches neither selection).
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:307]: Hmm this partially answers what I was wondering about above. I'm confused as I expected all tests to have a suite on them. I see now the plan is more about lifecycle determining suite. Will think on this, it might work well as defined.

File: enhancements/testing/test-suites.md [hunk 30]
```diff
@@ -239,22 +267,86 @@ applies to OKE testing in the same way as OCP.
...
+- **Default `LifeCycle` is `Informing`.** If a test carries no explicit
+  `LifeCycle` tag, it is treated as `LifeCycle: Informing`. Tests are
+  never automatically moved *out* of `Draft` — a test owner does that by
+  hand when the test is ready.
+- **`active` holds `Informing` and `Blocking`; `stable` holds only
+  `Stable`.** Because graduation only happens after a test has become
+  `Blocking` and passed GA + 1, `stable` never contains `Informing`
+  tests.
+- **`Criticality: Core` is additive.** It marks a test as a core smoke
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:316]: Interesting, so minimal is a subset, but will still run in active/stable suites/jobs. I like that.

File: enhancements/testing/test-suites.md [hunk 31]
```diff
@@ -239,22 +267,86 @@ applies to OKE testing in the same way as OCP.
...
+  test so it *also* appears in `conformance/*/minimal`. It does not
+  replace the `LifeCycle` membership: a `Core` test still lives in an
+  `active` or `stable` shard according to its `LifeCycle` (see
+  [Minimal-Suite Membership](#minimal-suite-membership)).
+
+The `LifeCycle` tag and the OTE `Informing()` annotation express the
+same fact for the non-blocking case and are kept in sync: `Informing()`
+present ⇔ `LifeCycle: Informing`; `Informing()` absent ⇔
+`LifeCycle: Blocking` or `Stable`.
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:325]: I wonder if we can get coderabbit to enforce that the labels look right on all tests across all repos.

File: enhancements/testing/test-suites.md [hunk 32]
```diff
@@ -277,135 +369,178 @@ yet fully defined** and remains an open question (see
...
+// Tag tests with structured metadata. Tags are key=value pairs set via
+// AddTag and queried in CEL as test.tags.<key>=="<value>".
+specs.Select(et.NameContains("[sig-foo] my test")).AddTag(map[string]string{
+    "LifeCycle": "Informing", // non-blocking during stabilization -> active
 })
 
-// Mark new tests as non-blocking during stabilization
+// Mark new tests as non-blocking during stabilization. The Informing()
+// annotation and LifeCycle: Informing tag are kept in sync.
 g.It("should do the thing", ote.Informing(), func() { ... })
```
Comment (dgoodwin) [inline enhancements/testing/test-suites.md:429]: can you include an example of a g.It with the tags for criticality and lifecycle? I'm nervous about the NameContains approach duplicating test names, then getting out of sync, I'd love examples of what a self-contained idea test definition looks like.

File: enhancements/testing/test-suites.md (hunk 22 above)
Comment (ironcladlou) [reply enhancements/testing/test-suites.md:168]: Nice, thanks

File: enhancements/testing/test-suites.md (hunk 25 above)
Comment (neisw) [reply enhancements/testing/test-suites.md:476]: We have talked about test metadata needs but not how they get populated.  ACK regarding the desired index, OTE source was my first thought when seeing this so that is a good place to start.
